src/protocol/thread/thread.ts

```diff
@@ -336,6 +341,32 @@ class _ThreadFront {
     }
   }
 
+  async searchSources(
+    { query }: { query: string },
+    onMatches: (matches: SearchSourceContentsMatch[]) => void
+  ) {
+    if (!this.searchWaiters) {
+      this.searchWaiters = new Map();
+      client.Debugger.addSearchSourceContentsMatchesListener(
+        ({ searchId, matches }: { searchId: string; matches: SearchSourceContentsMatch[] }) => {
+          const searchWaiter = this.searchWaiters?.get(searchId);
+          if (searchWaiter) {
+            searchWaiter(matches);
+          }
+        }
+      );
+    }
```

Comment:
I put this lazy-initialization code inline to keep the search logic co-located. I've since noticed that that's a bit of a departure from the way other ThreadFront things are initialized though, so I can totally change that.

Reply:
yeah - i'd probably do it at the top to be consistent. there isn't a perf hit